api/extension-guides/notebook.md

```diff
@@ -312,49 +320,46 @@ const GithubIssues: FunctionComponent<{ issues: GithubIssue[]; }> = ({ issues })
 
-If you have elements outside of the container or other asynchronous processes, you can use `onWillDestroyOutput` to tear them down. This event will fire when output is cleared, a cell is deleted, and before new output is rendered for an existing cell. For example:
+If you have elements outside of the container or other asynchronous processes, you can use `disposeOutputItem` to tear them down. This event will fire when output is cleared, a cell is deleted, and before new output is rendered for an existing cell. For example:
 
 ```js
 const intervals = new Map();
 
-notebookApi.onDidCreateOutput((evt) => {
-  render(<GithubIssues issues={evt.value} />, evt.element);
-
-  // create an interval that changes the color of the title <h2> every second:
-  intervals.set(evt.outputId, setInterval(() => {
-    evt.element.querySelector('h2').style.color = `hsl(${Math.random() * 360}, 100%, 50%)`;
-  }, 1000));
-});
+export const activate: ActivationFunction = (context) => ({
+    renderOutputItem(data, element) {
+        render(<GithubIssues issues={data.json()} />, element);
 
-notebookApi.onWillDestroyOutput(scope => {
-  if (scope === undefined) {
-    // If scope is undefined, all outputs will be destroyed:
-    for (const interval of intervals.values()) {
-      clearInterval(interval);
+        intervals.set(data.mime, setInterval(() => {
+            if(element.querySelector('h2')) {
+                element.querySelector('h2')!.style.color = `hsl(${Math.random() * 360}, 100%, 50%)`;
+            }
+        }, 1000));
+    },
+    disposeOutputItem(id) {
+        clearInterval(intervals.get(id));
+        intervals.delete(id);
     }
-    intervals.clear();
-  } else {
-    // Otherwise we're destroying a single output:
-    clearInterval(intervals.get(scope.outputId));
-    intervals.delete(scope.outputId);
-  }
 });
 ```
 
-It's important to bear in mind that all outputs for a notebook are rendered in different elements in the same iframe. If you use functions like `document.querySelector`, make sure to scope it to the specific output you're interested in to avoid conflicting with other outputs. In this example, we use `evt.element.querySelector` to avoid that issue.
+It's important to bear in mind that all outputs for a notebook are rendered in different elements in the same iframe. If you use functions like `document.querySelector`, make sure to scope it to the specific output you're interested in to avoid conflicting with other outputs. In this example, we use `element.querySelector` to avoid that issue.
+
+> **Note**: The following parts of Notebook API is still proposed and under development, which means some Notebook functionality, like the following, is only available on VS Code [Insiders](/insiders) and requires adding `vscode.proposed.d.ts` to your extension project. You can learn more in [Using Proposed APIs](/api/advanced-topics/using-proposed-api).
 
 ### Interactive Notebooks
```

Comment:
The content you have about the messaging API is useful, but it's actually not the right use case here--the original approach in this file remains the currently recommended way to do things.

The renderer messaging API is useful to allow for interactions between a specific renderer and the machine or filesystem. For example, the renderer might have a button that opens a folder in the file explorer. However, it _shouldn't_ be used to communicate directly to the controller like this, since that creates a hard dependency between the kernel and a specific renderer, and prevents other extension authors from being able to create renderers that work with the controller (or even at all, depending on how tight the coupling is). Instead, the controller preloads should be used to provide a API for all renderer extensions to use.

Reply:
I think I've reverted back to preload scripts but let me know if anything looks off!